Update: Sync and Connection Package Updates

sdixon194 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Backports updates to connection and sync packages

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
p1626184913242400-slack-C01U2KGS2PQ

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Proofread?

anomiex commented on projects/packages/identity-crisis/changelog/update-backport-packages-9.9:
```diff
@@ -0,0 +1,4 @@
+Significance: patch
```
This file probably shouldn't be being created. That's why things are wanting to update it to 0.2.2-alpha.

The removal of update-manager_get_connected_users should be enough to cover the updates going on in this PR.

sdixon194 replied:
Hmm, the tests were failing and looking for a changelog file for that. Should I undo the last commit where I updated the versions? I was trusting the tests 😬

anomiex commented:
Looks like https://github.com/Automattic/jetpack/runs/3058613966?check_suite_focus=true passed, the problem is the part where your PR is trying to downgrade jetpack-sync in projects/packages/backup/composer.json.

sdixon194 commented:
> Looks like Automattic/jetpack/runs/3058613966?check_suite_focus=true passed, the problem is the part where your PR is trying to downgrade jetpack-sync in projects/packages/backup/composer.json.

Yeah, that test failed before I added the changelog file to `identity-crisis`, then it was complaining about the version, so I updated that (evidently, incorrectly).

What version do we need `jetpack-sync` to be in its composer.json? I'm honestly unsure what's supposed to be what here :/

anomiex commented:
> Yeah, that test failed before I added the changelog file to identity-crisis, then it was complaining about the version, so I updated that (evidently, incorrectly).

https://github.com/Automattic/jetpack/runs/3058613966?check_suite_focus=true is the run against 7507e20fdd2dea140d3068beb8aa569cac120fbb, before you added that file.

> What version do we need jetpack-sync to be in packages/backup's composer.json? 1.24? (even though the packages/sync looks like it's 1.23.2?) I'm honestly unsure what's supposed to be what here :/

packages/sync shows 1.24.x at [composer.json line 58](https://github.com/Automattic/jetpack/blob/7507e20fdd2dea140d3068beb8aa569cac120fbb/projects/packages/sync/composer.json#L58). That's the version you need to be looking at.

What's happening here is that we released sync 1.23.1, then made three changes in master, then released a 1.23.2 with only one of those three changes. Master still contains the other two changes, so it's not itself at 1.23.2.

sdixon194 commented:
@anomiex seems it wants a changelog file again :(

anomiex commented:
Ah, I see. The "Linting / Changelogger use" is happy, but the gardening bot-comment is confused about it. @jeherve Any ideas on that?

We can ignore the confused gardening comment for now, although we'll probably want to fix that at some point.

jeherve commented:
>  the gardening bot-comment is confused about it.
> 
> We can ignore the confused gardening comment for now, although we'll probably want to fix that at some point.

It seems the gardening task expected a changelog entry since both `projects/packages/identity-crisis/package.json` and `projects/packages/identity-crisis/src/class-identity-crisis.php` were modified in the PR. 

Should we expand the check to ignore changes that only touch the version number in those files?

anomiex commented:
The existing "Changelogger use" check is satisfied by the _removal_ of the change entry files when making a release. It would also be satisfied by an edit to an existing change entry file. That's all potentially gameable if someone really wanted to avoid creating a change file, but I'm hoping that we don't have to worry about that (the check is mainly to catch "forgot to add an entry", not "actively avoiding adding an entry").

Maybe the gardening script should copy that logic?

anomiex commented:
Ah, I see what's going on. It does use that same logic ("any touched file counts"), but https://api.github.com/repos/Automattic/jetpack/pulls/20345/files is reporting some of the deletions as renames, like
```json
  {
    "sha": "c47cb18e829979a99c091812ea613db14cec2dac",
    "filename": "projects/packages/backup/changelog/jetpack-branch-9.9",
    "status": "renamed",
    "additions": 0,
    "deletions": 0,
    "changes": 0,
    "blob_url": "https://github.com/Automattic/jetpack/blob/1c26d09700d623e9278cc0bc8a8bae4ceecdd8ae/projects/packages/backup/changelog/jetpack-branch-9.9",
    "raw_url": "https://github.com/Automattic/jetpack/raw/1c26d09700d623e9278cc0bc8a8bae4ceecdd8ae/projects/packages/backup/changelog/jetpack-branch-9.9",
    "contents_url": "https://api.github.com/repos/Automattic/jetpack/contents/projects/packages/backup/changelog/jetpack-branch-9.9?ref=1c26d09700d623e9278cc0bc8a8bae4ceecdd8ae",
    "previous_filename": "projects/packages/identity-crisis/changelog/update-manager_get_connected_users"
  },
```
So `getFiles()` reports that `projects/packages/backup/changelog/jetpack-branch-9.9` was changed but not `projects/packages/identity-crisis/changelog/update-manager_get_connected_users`.
